pytest_fixtures/component/leapp.py

```diff
@@ -0,0 +1,16 @@
+# LEAPP Fixtures
+import pytest
+
+
+@pytest.fixture(scope="function")
+def check_pre_upgrade(target_sat):
+    target_sat.execute('yum update')
+    target_sat.execute('reboot')
```

Comment:
This is something that requires more effort to handle correctly.  Luckily, we have a helper method for that!
https://github.com/SatelliteQE/robottelo/blob/master/robottelo/hosts.py#L202

Reply:
Thanks @JacobCallahan I will check and refacotor this. There will be more additions to this PR.

> @shweta83 will you have a chance to revisit this before we branch in the next few weeks?

@JacobCallahan I will try to get this in before we branch.